### ISSUE TITLE
cannot find the option to change the picture related to an account 

### ISSUE THREAD

**Issue by vaibhavchellani** [MetaMask/metamask-extension#2471]
it would be great if we had the option of changing the display picture of each account .

**Comment by Zanibas** [comment]
We don't currently have this feature, but we have brought it up in the past. It's not currently high on our priority list, but we welcome PRs to implement this!

**Comment by vaibhavchellani** [comment]
@Zanibas sending PR , thanks !

**Comment by danfinlay** [comment]
Right now, the pictures are deterministically generated from the address's account. Ways we could approach this:

- We could allow users to locally set their pictures to other things (would get wiped in case of hard drive failure, and would not sync to other computers)
- We could use some kind of on-chain registry or just public API for users to back up their custom pictures.
- We could add new icon generation algorithms.
- When first generating a vault, we could let the user re-generate the account until they were fond of the image they were shown.

Which one are you thinking of?

**Comment by sergeiwallace** [comment]
I know this is a small thing but I'd definitely appreciate it!

**Comment by dairefagan** [comment]
This would help quickly identify one account of many, and also act as another visual confirmation of the correct account.

**Comment by Go-Merk** [comment]
I think it would be nice for user to use their NFT as profile pic, any kind of image (PNG, JPG, WEBP, SVG... I think they are the most common types)

**Comment by lunesco** [comment]
Yes, it would be great to be able to set your own image as an avatar in MetaMask.

**Comment by chrismattmann** [comment]
@vaibhavchellani I also agree setting the account avatar image would be a great feature.

**Comment by actuallymentor** [comment]
Low hanging fruit would be to integrate ENS avatars.

**Comment by dustinkeeton** [comment]
@actuallymentor not only low hanging fruit but this would be my preferred solution.

**Comment by michaelpayne02** [comment]
Can we please get ENS avatars implemented? Many sites already support displaying avatars and more are constantly adding it.
